doc/request-signing.md

```diff
@@ -0,0 +1,46 @@
+# Request signing
+
+## Canonical request form
+
+The request is first transformed into a canonical form for signing,
+which is constructed as follows (in pseudo-Haskell):
+
+```haskell
+r_canonical = intercalate "\n" [
+    requestMethod
+  , canonicalURI
+  , canonicalQueryString
+  , canonicalHeaders
+  , signedHeaders
+  , hashedPayload
+  ]
+  where
+    hashedPayload = hexDigest $ sha256 requestPayload
+```
+
+FIXME: finish this section/specify the rest properly
+
+## Signed data
+
+The following data is included in the request's signature:
+
+ - algorithm
+ - request date
+ - request scope
```

Comment:
I'm not sure yet, was one of the things I was hoping to talk to you about. Was thinking something like `API/service + key_id`.